### ISSUE TITLE
es6: Template Strings

### ISSUE THREAD

**Issue by hhsnopek** [jshint/jshint#2238]
The template strings within jshint have errors being thrown at them in 2 cases:

1. ```Unexpected '`'.``` after `label:`
```javascript
_menu.append(new _gui.MenuItem({
  label: `${_repo.repoName}`,
  icon: 'img/gh-repo.png'
}));
```

2. Same error occurs here: ```Unexpected '`'```
```javascript
_save(
  `${_repo.repoName}`,
  `#${_issue.issue} - ${_issue.title}`
);
```

Note: I'm using jshint plugin for atom

**Comment by caitp** [comment]
Can you provide a bit more information?

1. what version of jshint is atom using?
2. what are the enforcement/relaxation settings it's using?

**Comment by hhsnopek** [comment]
[atom-jshint](https://github.com/sindresorhus/atom-jshint)
It uses `^2.4.4`, default settings found here: [.jshintrc](https://github.com/sindresorhus/atom-jshint/blob/master/.jshintrc) 

I suppose this might want to be opened on atom-jshint as well?

I also just found when using let I get this: `'let' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).`. I'm thinking I need to enable this in my `.jshintrc` now

**Comment by caitp** [comment]
The first work on template literals was added in https://github.com/jshint/jshint/commit/b4be350f6720b1a824f76fc60e383a6c426da69c (v2.5.0), and has been improved incrementally since, with some major fixes in the past 3 patch releases.

**Comment by caitp** [comment]
`^2.4.4` should pull in the new updates though... so, maybe you need to reinstall the modules?

**Comment by hhsnopek** [comment]
I just uninstalled atom-jshint, cloned it down, updated modules, and `apm link`ed it in, same issues as before :frowning:

**Comment by caitp** [comment]
With the esnext option set to true?

**Comment by hhsnopek** [comment]
yeah, I added `"esnext": true` to atom-jshint's `.jshintrc` with no change

**Comment by hhsnopek** [comment]
cc @sindresorhus

**Comment by hhsnopek** [comment]
Here we go, I added an `.jshintrc` to my project, which I then started to work properly

**Comment by HamsterCoder** [comment]
I am currently struggling with a similar issue:
```
{
     'data-bind': `text:
          (YT.pageTitle.path() ? YT.pageTitle.path() + ' - ' : '') +
          (YT.pageTitle.page() ? YT.pageTitle.page() + ' - ' : '') +
           YT.pageTitle.cluster`
}
```

The beginning of the template string is marked with error  - ```Unexpected '`' E024```

*  I have latest ```Webstorm 11.0.3```  with built-in jshint plugin ```version 2.1.10```
*  I have ```esnext: true``` option in my .jshintrc (For example it is not highlighting ```let a = 'foo'```)
*  And it seems to lint fine using ```jshint-groups 0.8.0``` configured similarly.

Honestly I don't know what to ask - I'll just leave this info here.

**Comment by HamsterCoder** [comment]
There is another problem - I am trying to ignore code altogether using 
```/* jshint ignore:start */```
```/* jshint ignore:end */```
found in docs here http://jshint.com/docs/ and get an error E001 ```Bad option:  'ignore'```.

**Comment by jugglinmike** [comment]
> *  I have latest ```Webstorm 11.0.3```  with built-in jshint plugin ```version 2.1.10```
> *  I have ```esnext: true``` option in my .jshintrc (For example it is not highlighting ```let a = 'foo'```)
> *  And it seems to lint fine using ```jshint-groups 0.8.0``` configured similarly.

These projects are not maintained by the JSHint team, so I can't diagnose the problem from this information alone. It would be helpful if you could share the version of JSHint that they include.

**Comment by HamsterCoder** [comment]
https://github.com/ikokostya/jshint-groups uses "jshint": "~2.6.0" and works fine.

As for Webstorm I think their plugin version corresponds to jshint version somehow if not exactly. https://www.jetbrains.com/webstorm/help/jshint.html

And thankfully I found the source of my problem - I didn't see the scroll, when I did - I update plugin to version 2.8.0 and it started working. Sorry for your time.
